src/Features/CSharp/Portable/CodeRefactorings/InlineMethod/CSharpInlineMethodRefactoringProvider.cs

```diff
@@ -0,0 +1,194 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Composition;
+using System.Linq;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.LanguageServices;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.InlineMethod;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeRefactorings.InlineMethod
+{
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = nameof(PredefinedCodeRefactoringProviderNames.InlineMethod)), Shared]
+    [Export(typeof(CSharpInlineMethodRefactoringProvider))]
+    internal sealed class CSharpInlineMethodRefactoringProvider :
+        AbstractInlineMethodRefactoringProvider<InvocationExpressionSyntax, ExpressionSyntax, MethodDeclarationSyntax, StatementSyntax>
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public CSharpInlineMethodRefactoringProvider() : base(CSharpSyntaxFacts.Instance, CSharpSemanticFactsService.Instance)
+        {
+        }
+
+        protected override ExpressionSyntax? GetInlineExpression(MethodDeclarationSyntax methodDeclarationSyntax)
+        {
+            var blockSyntaxNode = methodDeclarationSyntax.Body;
+            if (blockSyntaxNode != null)
+            {
+                // 1. If it is an ordinary method with block
+                var blockStatements = blockSyntaxNode.Statements;
+                if (blockStatements.Count == 1)
+                {
+                    var statementSyntax = blockStatements[0];
+                    return statementSyntax switch
+                    {
+                        // Note: For this case this will return null in Callee()
+                        // void Caller() { Callee(); }
+                        // void Callee() { return; }
+                        // Refactoring won't be provided for this case.
+                        ReturnStatementSyntax returnStatementSyntax => returnStatementSyntax.Expression,
+                        ExpressionStatementSyntax expressionStatementSyntax => expressionStatementSyntax.Expression,
+                        _ => null
```

Comment:
we should also consider `throw statement/expression`

Reply:
i say that because that's what our other 'single statement method' features work upon.

Reply:
OK, I see that the main scenario would be throw statement.
I can now image only one scenario. Callee() is Expression Statement
`void Caller() { Callee(); }`
`int Callee() { throw new Exception(); }`
After inline it should be
`void Caller() { throw new Exception(); }`
Other than that, like 
`void Caller() { DoSomething(Callee()) }`
`int Callee() { throw new Exception(); }`
Becuase `throw new Exception()` cant be used as expression. So in this scenario refactoring should not be provided.

For throw expression, because it is usually used in condtional and null-coalecing expression, and these expression will be  used in ReturnStatement/ExpressionStatement, so I would assume it is supported already. I can add some tests for that.

Reply:
interesting points :)

Reply:
Add the support & test for this part.